src/main/java/net/logstash/logback/stacktrace/ThrowableHasher.java

```diff
@@ -0,0 +1,120 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.logstash.logback.stacktrace;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * Utility class that generates a hash for any Java {@link Throwable error}
+ *
+ * @author Pierre Smeyers
+ */
+public class ThrowableHasher {
+
+    private final StackElementFilter filter;
+
+    /**
+     * Constructs a {@link ThrowableHasher} with the given filter
+     *
+     * @param filter filter
+     */
+    public ThrowableHasher(StackElementFilter filter) {
+        this.filter = filter;
+    }
+
+    /**
+     * Constructs a {@link ThrowableHasher} using {@link StackElementFilter#any()} filter
+     */
+    public ThrowableHasher() {
+        this(StackElementFilter.any());
+    }
+
+    /**
+     * Generates a Hexadecimal hash for the given error stack
+     * <p>
+     * Two errors with the same stack hash are most probably same errors
+     */
+    public String hexHash(Throwable error) {
+        return hexHashes(error).peek();
+    }
+
+    /**
+     * Generates and returns Hexadecimal hashes for the error stack and each ancestor {@link Throwable#getCause() cause}
+     * <p>
+     * The first queue element is the stack hash for the topmost error, the next one (if any) is it's direct
+     * {@link Throwable#getCause() cause} hash, and so on...
+     */
+    public Deque<String> hexHashes(Throwable error) {
+        Deque<String> hexHashes = new ArrayDeque<String>();
+        hash(error, hexHashes);
+        return hexHashes;
+    }
+
+    /**
+     * Generates a hash (int) of the given error stack
+     * <p>
+     * Two errors with the same stack hash are most probably same errors
+     */
+    int hash(Throwable error, Deque<String> hexHashes) {
+        int hash = 0;
+
+        // compute parent error hash
+        if (error.getCause() != null && error.getCause() != error) {
+            // has parent error
+            hash = hash(error.getCause(), hexHashes);
+        }
+
+        // then this error hash
+        // hash error classname
+        hash = 31 * hash + error.getClass().getName().hashCode();
+        // hash stacktrace
+        for (StackTraceElement element : error.getStackTrace()) {
+            if (accept(element)) {
+                hash = 31 * hash + hash(element);
+            }
+        }
+
+        // push hexadecimal representation of hash
+        hexHashes.push(String.format("%08x", hash));
+
+        return hash;
+    }
+
+    boolean accept(StackTraceElement element) {
+        // skip null element, generated class or filter element
+        return element != null && element.getFileName() != null && element.getLineNumber() >= 0 && filter.accept(element);
```

Comment:
I think the `element.getFileName() != null && element.getLineNumber() >= 0` check will cause classes that weren't generated with debug information to not be evaluated. i.e. only classes compiled with debug info have the file/line number info.

If you're ok with that, then that's fine.  But probably worth documenting.

If you're not ok with that, then you'll need to remove the check, and just rely on the user to configure the filter to remove dynamically generated stack frames.